components/automate-gateway/gateway/middleware/authv2/authv2.go

```diff
@@ -54,65 +54,31 @@ func (c *client) Handle(ctx context.Context, subjects []string, projectsToFilter
 		"iam.version":   "iam_v2",
 	})
 
-	var projects []string
-	// TODO (TC) Needs feature flag work to be implemented before
-	// we can toggle between 2.0 and 2.1 aka enable projects AuthZ.
-
-	// if minorVersion == "0" {
-
 	// Note: if ANYTHING goes wrong, 403 is the error we return. This is done
 	// on purpose, so our authz response doesn't leak information about what
 	// is happening internally.
-	resp, err := c.client.IsAuthorized(ctx, &authz.IsAuthorizedReq{
-		Subjects: subjects,
-		Resource: resource,
-		Action:   action,
+	filteredResp, err := c.client.ProjectsAuthorized(ctx, &authz.ProjectsAuthorizedReq{
+		Subjects:       subjects,
+		Resource:       resource,
+		Action:         action,
+		ProjectsFilter: projectsToFilter,
 	})
 	if err != nil {
 		if status.Convert(err).Code() == codes.FailedPrecondition {
 			return nil, err
 		}
 		log.WithError(err).Error("error authorizing request")
 		return nil, status.Errorf(codes.PermissionDenied,
-			"error authorizing action %q on resource %q for subjects %q: %s",
-			action, resource, subjects, err.Error())
+			"error authorizing action %q on resource %q for subjects %q filtered by project list %q: %s",
```

Comment:
question: should we drop references to projects in these error messages so they don't leak into v2 responses? we could also maybe push down the error handling into ProjectsAuthorized so it returns different errors based on the version.

Reply:
The former at least (dropping references). The latter would be nice, though.

Reply:
i'll drop it for now 👍